pyproject.toml

```diff
@@ -41,23 +41,55 @@ classifiers = [
 [tool.poetry.dependencies]
 python = "^3.7"
 PyYAML = "^6.0"
-pylink-square = ">=0.12,<0.15"
-pyserial = "^3.0"
-python-can = {version = "^4.0.0", extras = ["pcan"]}
-PyVISA = "^1.12.0"
-PyVISA-py = "~0.5.3"
 robotframework = "3.2.2"
 unittest-xml-reporting = "^3.2.0"
 click = ">=7.0.0,<9.0.0"
-pykiso-python-uds = "~3.0.2"
 tabulate = ">=0.8.9,<0.10.0"
 Jinja2 = ">=2.11.0,<4.0.0"
 MarkupSafe = "~2.0.1" # Allow support for Jinja2 between 2.11 < x < 3
 importlib-metadata = {version = ">=4.12,<6.0", python = "<3.8"}
 pyreadline3 = {version = "^3.4.1", python = "^3.5"}
+pylink-square = {version = ">=0.12,<0.15", optional = true}
+pykiso-python-uds = {version = "~3.0.2", optional = true}
+pyserial = {version = "^3.0", optional = true}
+PyVISA = {version = "^1.12.0", optional = true}
+PyVISA-py = {version = "~0.5.3", optional = true}
+python-can = {version = "^4.0.0", optional = true, extras = ["pcan,vector"]}
 
+[tool.poetry.extras]
+plugins = [
+    "pylink-square",
+    "pykiso-python-uds",
+    "python-can",
+    "pyserial",
+    "PyVISA",
+    "PyVISA-py",
+]
+can = [
+    "pykiso-python-uds",
+    "python-can",
+]
+debugger = [
+    "pylink-square",
+]
+instrument = [
+    "PyVISA",
+    "PyVISA-py",
+]
+serial = [
+    "pyserial",
+]
+all = [
+    "pylink-square",
+    "pykiso-python-uds",
+    "python-can",
+    "pyserial",
+    "PyVISA",
+    "PyVISA-py",
+]
```

Comment:
Is it possible to re-use variables with poetry ?
eg:
all = can + debugger + instrument + serial

what is the difference between 'all' and 'plugins' ?

Reply:
Hey!
So @Pog3k answered your question :)
For the last point. Next step would be to separate the different tools from the core part. Which mean that there will be a difference between `plugins` and `all`.